### PR TITLE
chore: Update readme [skip ci]

### PR DESCRIPTION
**Pull request by jouni**
Simplify and cleanup the readme, try to make it easier to digest.

Cosmetic updates to BUILD.md

[Preview](https://github.com/vaadin/platform/blob/update-readme/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,159 @@
...
-|------|------------|
-| Designer | https://github.com/vaadin/designer |
-| TestBench | https://github.com/vaadin/testbench |
+## Integrations
+Plugins and add-ons for easy integration with 3rd party development tools.
 
-All tools are part of the [Vaadin Pro Subscription](https://vaadin.com/pricing).
+- [**Spring** add-on](https://github.com/vaadin/spring)
+- [**CDI** add-on](https://github.com/vaadin/cdi)
+- [**Maven** plugin](https://github.com/vaadin/maven-plugin)
```
Comment (ZheSun88) [inline README.md:101]: this link is for the maven-plugin for FW 7 and 8.

the code part of maven plugin has been moved into flow, which artifact name is `flow-maven-plugin`, when releasing with platform, we repackage everything with artifact name `vaadin-maven-plugin`

File: README.md [hunk 2]
```diff
@@ -1,70 +1,159 @@
...
+- [**Gradle** plugin](https://github.com/vaadin/vaadin-gradle-plugin)
```
Comment (ZheSun88) [inline README.md:102]: we support Gradle plugin for Vaadin 20+ (currently for V21), this part of code has been included in flow project too, same with maven plugin. 
this link is for the gradle-plugin for platform 14.x, which has not been announced as official support for Vaadin.

File: README.md (hunk 1 above)
Comment (jouni) [reply README.md:101]: Okay, thanks for the clarification!

I suppose there’s no place to link to then? Would be nice to show Maven support here somehow, but I’m not sure now how to best do that. But I suppose this is related to the Gradle support in a way, as they are both deps and build tools.

Comment (jouni) [reply README.md:101]: Would this be good enough?

```suggestion
- [**Maven** plugin](https://github.com/vaadin/flow/tree/master/flow-plugins/flow-maven-plugin)
```

File: README.md (hunk 2 above)
Comment (jouni) [reply README.md:102]:
```suggestion
- [**Gradle** plugin](https://github.com/vaadin/flow/tree/master/flow-plugins/flow-gradle-plugin)
```

File: README.md (hunk 1 above)
Comment (jouni) [reply README.md:101]: I assume it’s at least better than linking to the old plugin 😄

File: README.md [hunk 3]
```diff
@@ -1,70 +1,159 @@
-# Vaadin platform
+<div align="center">
 
-[Vaadin platform](https://vaadin.com/) consists of a set of web components, a Java web framework called Vaadin Flow, configurable themes, tools and a set of opinionated app starters. Platform releases gather all products in to a single package every quarter. The product versions included in a platform release are not necessarily the latest released versions, but are hand picked so that they work well together.
+![Vaadin](vaadin-banner.svg)
 
-Visit [vaadin.com](https://vaadin.com/) to learn more about Vaadin.
+Vaadin is a web app development platform for Java. It helps you build reliable web apps with great UX faster than before.
```
Comment (web-padawan) [inline README.md:5]: Let's consider adding a line break here to make the description look more readable.

### Before

Screenshot 2021-11-22 at 12 23 47

### After

Screenshot 2021-11-22 at 12 24 19

Comment (jouni) [reply README.md:5]: Yeah, that looks nicer at least on large viewport sizes. I suppose that’s more common than a small viewport

Screen Shot 2021-11-22 at 12 28 37

Comment (web-padawan) [reply README.md:5]: Thanks, that makes sense. Let's keep it then.

**Review by web-padawan — APPROVED** [review]

**Comment by jouni** [comment]
I think I’ll work on moving most of this content to the github.com/vaadin readme instead, and let the readme of this repo be more about what the purpose of this repo technically is.

**Pull request merged**